pyqtgraph/graphicsItems/InfiniteLine.py

```diff
@@ -240,3 +362,122 @@ def setMouseHover(self, hover):
+        elif self.angle == 0:  # horizontal line
+            diffY = ymax-ymin
+            diffMin = self.value()-ymin
+            limInf = self.shift*(ymax-ymin)
+            xpos = xmin+self.location*(xmax-xmin)
+            if diffMin < limInf:
+                self.text.anchor = Point(self.anchorUp)
+            else:
+                self.text.anchor = Point(self.anchorDown)
+            fmt = " y = " + self.format
+            if self.unit is not None:
+                fmt = fmt + self.unit
+            self.text.setText(fmt.format(self.value()), color=self.textColor)
+            self.text.setPos(xpos, self.value())
+
+
+    def showLabel(self, state):
+        """
+        Display or not the label indicating the location of the line in data
+        coordinates.
+
+        ==============   ==============================================
+        **Arguments:**
+        state            If True, the label is shown. Otherwise, it is hidden.
+        ==============   ==============================================
+        """
+        if state:
+            self.text.show()
+        else:
+            self.text.hide()
+        self.update()
+
+    def setLocation(self, loc):
+        """
+        Set the location of the textItem with respect to a specific axis. If the
+        line is vertical, the location is based on the normalized range of the
+        yaxis. Otherwise, it is based on the normalized range of the xaxis.
+
+        ==============   ==============================================
+        **Arguments:**
+        loc              the normalized location of the textItem.
+        ==============   ==============================================
+        """
+        if loc > 1.:
+            loc = 1.
+        if loc < 0.:
+            loc = 0.
+        self.location = loc
+        self.update()
+
+    def setShift(self, shift):
+        """
+        Set the value with respect to the normalized range of the corresponding
+        axis where the location of the textItem shifts from one side to another.
+
+        ==============   ==============================================
+        **Arguments:**
+        shift              the normalized shift value of the textItem.
+        ==============   ==============================================
+        """
+        if shift > 1.:
+            shift = 1.
+        if shift < 0.:
+            shift = 0.
+        self.shift = shift
+        self.update()
+
+    def setFormat(self, format):
+        """
+        Set the format of the label used to indicate the location of the line.
+
+
+        ==============   ==============================================
+        **Arguments:**
+        format           Any format compatible with the new python
+                         str.format() format style.
+        ==============   ==============================================
+        """
+        self.format = format
+        self.update()
+
+    def setUnit(self, unit):
```

Comment:
For consistency with the rest of the library, consider `suffix` instead of `unit`.

I am also wondering if this is necessary, given the availability of `setFormat` ? There's a general problem here that exists throughout the library that I have not yet found a good solution for:  how to let the user specify text formatting for numerical values (for example, in AxisItem, SpinBox, and ValueLabel). In most places, we use `siFormat` to generate the strings, but it would be even nicer to expose this via some string formatting syntax like `str.format()` uses. Imagine my line is at x=0.001, and I want the label to read "val: 1 mm":

    line.setFormat('val: {value}m')  # unscaled units
    line.setFormat('val: {siScaledValue}m')   # automatically scaled to 1 mm
    
Those two possibilities would be easy enough to support, but anything beyond that would be too difficult and perhaps just requires the user to provide a callable to generate the string. Perhaps `setFormat(callable)` should also be allowed? I don't know..